error.go

```diff
@@ -0,0 +1,921 @@
+	Unauthenticated Code = 16
+)
+
+// Visibility controls information disclosure across trust boundaries
+type Visibility int
+
+const (
+	// VisibilityInternal - error must not be shown to clients (most restrictive)
+	VisibilityInternal Visibility = 0
+	// VisibilityPrivate - can be shared within trusted organization boundaries
+	VisibilityPrivate Visibility = 1
+	// VisibilityPublic - safe for any client including end-users (least restrictive)
+	VisibilityPublic Visibility = 2
+)
+
+
+// HelpLink provides documentation link
+type HelpLink struct {
+	// description of what the link offers
+	description string
+	// url must be absolute and publicly accessible
+	url string
+}
+
+// Help provides links to relevant documentation
+type Help struct {
+	// links pointing to additional information on handling the error
+	links []HelpLink
+}
+
+// MetadataValue contains both the value and its visibility level
+type MetadataValue struct {
+	// value is the actual metadata value
+	value string
+	// visibility determines whether this field should be exposed based on trust boundary
+	visibility Visibility
+}
+
+// Metadata represents a map of metadata with visibility control
+type Metadata = map[string]MetadataValue
+
+// DebugInfo contains technical details for internal debugging
+type DebugInfo struct {
+	// stackEntries indicating where the error occurred
+	stackEntries []string
+	// detail provides contextual information about what the system was attempting
+	detail string
+}
+
+// LocalizedMessage provides translated error message
+type LocalizedMessage struct {
+	// locale follows BCP 47 (e.g., "en-US", "fr-CH")
+	locale string
+	// message in the specified locale
+	message string
+}
+
+
+// RetryInfo describes when a client can retry a failed request
+// Following ADR requirements: servers MUST set either retry_offset OR retry_time, never both
+type RetryInfo struct {
+	// retryOffset specifies a duration to wait before retrying
+	retryOffset *time.Duration
+	// retryTime specifies an absolute timestamp after which to retry
+	retryTime *time.Time
+}
+
+// TrogonError represents the standardized error format following the ADR
+type TrogonError struct {
+	// specVersion identifies the version of the error specification
+	specVersion int
+	// code provides standardized error category
+	code Code
+	// message contains generic description in English (nil means use code's default)
+	message *string
+	// domain identifies the context in which an error happened
+	domain string
+	// reason identifies the proximate cause of the error
+	reason string
+	// metadata provides additional structured details with visibility control
+	metadata Metadata
+	// causes contains composed errors or error wrapping scenarios
+	causes []*TrogonError
+	// visibility controls information disclosure
+	visibility Visibility
+
+	// Optional fields
+	// subject identifies the specific entity that caused the error
+	subject *string
```

Comment:
why pointers to strings ?  (empty string should be fine/same as nil, without NPE)

Reply:
disambiguate “unset” vs “empty”, what do you recommend me here? I can make the Getter to be a pointer I guess. 

What do you recommend me to do?

Reply:
I recommend that for most case unset or empty is same/doesn't matter when the value is a string, if you have a string it's fairly rare that "" needs to be explictly set and different from not setting it